source/hash_table.c

```diff
@@ -212,7 +212,7 @@ static int s_update_template_size(struct hash_table_state *template, size_t expe
     }
 
     template->size = size;
-    template->max_load = (size_t)(template->max_load_factor * template->size);
+    template->max_load = (size_t)(template->max_load_factor * (double)template->size);
```

Comment:
@bdonlan this change is making me a little uncomfortable.
I don't know how big the template->size can be, but if it gets big enough (more than 63 bits), I think this will be a bug when converting to a signed floating point.

I'm debating whether we should multiply the max_load_factor by 100 and cast that to size_t instead (and obviously divide template->size by 100 to even it out)
Thoughts?